lich.rbw

```diff
@@ -10760,6 +10756,143 @@ if defined?(Gtk)
    Gtk.queue { Gtk::Window.default_icon = Gdk::Pixbuf.new(Zlib::Inflate.inflate("eJyVl8tSE1EQhieTggULX8utr2CVOzc+gpXMhUDIBYhIAuQCAQEDKDEiF9eWZVneUDc+gk+gMtKN+doTJ1pSEP45/Z/uPn07k+u3bt/wvGsLfsbzPfn1vLvyl9y843n68Vw+cpdryYWLMoIS0H9JFf0QlAelSQNB3wVFgr6B/r6WJv2K5jnWXrDWFBQLmnWl6kFd0IygYop0SVARaSjoJagqqIxmXXuFlpKgiqAZ1l6juSCoBlLpG6IWC1p0fX7rSqvued9x3ozv+0kkj/P6ePmTBGxT8ntUKa8uKE+YRqQN1YL0gxkSzUrpyON5igdqLSdoWdAzKA3XRoAHyluhuDQ92V/WEvm/io5pdNjOOlZrSNdABfTqjlVBC4LmBFXwpMkOTXubdFbdZLddXpcoLwkK8WUdnu4dIC0SR9XXdXl9kPq3iVQL6pAS1Lw8cKWq9JADasC2OaBSmrihhXIqqEeYqiSsxwF1bSBoD+NqSP3ry6M/zNWugEWc11mjhdGisXSrhuBMHlWwTFiOCb2Ww6ygI9aa9O2AtTYBV80ajEPqc4MdBxjvEPWCoH0yqzW7ImgL6UO3olcI6bSgXfYG9JTa2CExOXnMjSVPTExc2ci5R9jGbgi558TaaCElWMflEHVbaNLeKSGYxSkrxk3IM+jTs2QmJyevJormc4MQzDIuzJd1wldEqm5khw3NAdokqER8uzjRogjKGFRnFzETUhhFynONvRGzYA6plVfEbCGqEVmJ2NChRmM2xAyUkH7rssOSGeNASDJtx4EgP5vNJjRUk361IakddcahTBoiPRHUGSc9cgdN7GrWgDipUU/84crOOLUt2nis9DRFWsPoyThpndYJ4ZnraVpGDnbPtLCmCTkep0qltiNkFAYpmhukP2BEBa7SNN6Aqhvh1SkO5fUpMbO7lMJ7DK+Q4p8Vm824OZenzWwXUhGTEZMkj7Tk+nLVKlK8EXMhT+dP45pdUlXXeEywI0ogYqbY5ZijpPKu8cANZ8ABQ7ohQ37N8dC1toz0qevfSKIaSJ/gi/EKKby+yws5W8AcUfTI7UfVZ++e9iqjd1h2amqKF6MafEusvcjsU9c5V7llsgV5D7QAr8xaB9QDzcOrsKYzWYvC7jy71RRV2FZlm+V5F9fK8Obdc2xC3nHrOsL7HIVnL0F2rcXw7BLNoM/SHhNi5Z2zprU+Q2JV+tFtFr217uOBrn2SR+2xq1fc30fuZzpHryaG7xfUrqHswmGMfBfzxbd/s4Z2U1jI7PteQgZGkVhL/txl3zV/AnftNz0=".unpack('m')[0]).unpack('c*'), false) }
 end
 
+module EAccess
+  PEM = File.join("#{DATA_DIR}/", "simu.pem")
+#  pp PEM
+  PACKET_SIZE = 8192
+
+  def self.pem_exist?
+    File.exist? PEM
+  end
+
+  def self.download_pem(hostname = "eaccess.play.net", port = 7910)
```

Comment:
I *really* don't like the idea of fetching the certificate we're going to use to authenticate/validate the connection to the same server.

Reply:
Note the way the internet normally handles self-signed certs is to just not validate at all, which is equally valid here, given the choices Simu made.

Reply:
Aside from gently asking Simu to do it a better way, I don't really have a better idea. Checking is better than not, even if not by much.

Reply:
We know what it is currently, and it's expiry is...long. I'm more of the mind of hard-coding it in (like done with lnet) for the time being.  

I've requested to the PM of DragonRealms that they should use the already existing CA signed wildcard cert they have for the website in it's place (we would inherit the CRL, the hostname validation, etc if they do that), but until that's done, my personal preference is validating to what we've been given and we know is currently valid.

Reply:
This (hardcoded cert check) is what is done with SGE currently btw, so presumably we'd have some notice if they change it in the form of an announcement and an SGE update.

Reply:
We could also include the file locally with any distributions and if it resorts to the download do the standard user confirmation spiel like when adding a fingerprint from a server you connect to. I can't see the world where they can hijack communication to the server but can't also download the publicly available cert to use so not sure how much effort it's worth worrying over for now.

Reply:
I'm leaving it as is. I think it's a reasonable approach given Simu's choices. Whenif Simu changes, we adapt to that.

Reply:
Briefly weighing in - don't care for it myself.  Given all the faults discussed, I'd leave it.  Two reasons.  
1) I don't have to rebundle lich if it gets changed, and 
2) we won't know if it gets revoked and reissued short of watching SGE / news - but we also don't care if a new and valid cert is available.